src/plugins/intel_gpu/src/graph/impls/cm/impl_example.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+
+#include "fully_connected_inst.h"
+#include "impls/registry/implementation_manager.hpp"
+
+namespace cldnn {
+namespace cm {
+
+struct ExampleImplementationManager : public ImplementationManager {
+    OV_GPU_PRIMITIVE_IMPL("cm::example")
+    ExampleImplementationManager(shape_types shape_type, ValidateFunc vf = nullptr)
+        : ImplementationManager(impl_types::cm, shape_type, vf) {}
+
+    std::unique_ptr<primitive_impl> create_impl(const program_node& node,
+                                                const kernel_impl_params& params) const override;
+
+    bool validate_impl(const program_node& node) const override {
+        assert(node.is_type<fully_connected>());
```

Comment:
I think CM impls must ensure that CM compiler is available on the system (I think on linux it's not guaranteed to be installed)

Reply:
I will try to implement this check inside kernel_selector_helper cache the result and query it from the ImplementationManager validate function.

Reply:
I added CM check in fa489c8. Check can be used from ImplementationManager to validate CM impls and also from tests to skip CM tests in case runner does not support it.